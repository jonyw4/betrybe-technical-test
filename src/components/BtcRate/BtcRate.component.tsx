import * as React from 'react';

export function BtcRate({ rate, title }: BtcRateProps) {
  return (
    <div className="btc-rate">
      <div className="btc-rate__title">{title}</div>
      <div className="btc-rate__rate">{rate}</div>
    </div>
  );
}

interface BtcRateProps {
  rate: string;
  title: string;
}
