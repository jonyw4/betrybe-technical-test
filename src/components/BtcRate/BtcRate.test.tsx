import React from 'react';
import { mount } from 'cypress-react-unit-test';
import { BtcRate } from './BtcRate.component';

describe('components > BtcRate', () => {
  it('should render a default BtcRate correctly', () => {
    mount(<BtcRate rate="10" title="Titulo" />);
    cy.get('.btc-rate').should('be.visible');
    cy.get('.btc-rate__title').contains('Titulo');
    cy.get('.btc-rate__rate').contains('10');
  });
});
