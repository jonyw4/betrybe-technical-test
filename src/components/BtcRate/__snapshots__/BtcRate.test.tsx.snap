// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > BtcRate should renders a BtcRate correctly 1`] = `
<div
  className="btc-rate"
>
  <div
    className="btc-rate__title"
  >
    BRL
  </div>
  <div
    className="btc-rate__rate"
  >
    10
  </div>
</div>
`;
